service/src/test/java/uk/gov/hmcts/reform/fpl/jobs/JobSchedulerTest.java

```diff
@@ -0,0 +1,161 @@
+
+        jobScheduler.scheduleCronJobs();
+
+        verify(scheduler, never()).deleteJob(any());
+        verify(scheduler, times(2)).scheduleJob(jobDetailsCaptor.capture(), triggerCaptor.capture());
+
+        List<JobDetail> jobs = jobDetailsCaptor.getAllValues();
+        List<Trigger> triggers = triggerCaptor.getAllValues();
+
+        assertJob(jobs.get(0), enabledSchedule1);
+        assertTrigger(triggers.get(0), enabledSchedule1);
+
+        assertJob(jobs.get(1), enabledSchedule2);
+        assertTrigger(triggers.get(1), enabledSchedule2);
+    }
+
+    @Test
+    void shouldNotCreateAnyJobIfAllSchedulesAreDisabled() throws Exception {
+        when(schedulerConfig.getSchedulerConf()).thenReturn(List.of(disabledSchedule));
+        when(scheduler.getJobKeys(any())).thenReturn(emptySet());
+
+        jobScheduler.scheduleCronJobs();
+
+        verify(scheduler, never()).deleteJob(any());
+        verify(scheduler, never()).scheduleJob(any(), any());
+    }
+
+    @Test
+    void shouldThrowExceptionWhenCronExpressionIsInvalid() throws Exception {
+        when(schedulerConfig.getSchedulerConf()).thenReturn(List.of(invalidSchedule));
+        when(scheduler.getJobKeys(any())).thenReturn(emptySet());
+
+        assertThrows(Exception.class, jobScheduler::scheduleCronJobs);
+    }
+
+    @Test
+    void shouldRecreateJobs() throws Exception {
+        when(schedulerConfig.getSchedulerConf()).thenReturn(List.of(enabledSchedule2));
+        when(scheduler.getJobKeys(any())).thenReturn(Set.of(jobKey(enabledSchedule1), jobKey(disabledSchedule)));
+
+        jobScheduler.scheduleCronJobs();
+
+        verify(scheduler).deleteJob(jobKey(enabledSchedule1));
+        verify(scheduler).deleteJob(jobKey(disabledSchedule));
+        verify(scheduler).scheduleJob(jobDetailsCaptor.capture(), triggerCaptor.capture());
+
+        assertJob(jobDetailsCaptor.getValue(), enabledSchedule2);
+        assertTrigger(triggerCaptor.getValue(), enabledSchedule2);
+    }
+
+    private static JobKey jobKey(SchedulerConfiguration.Schedule schedule) {
+        return JobKey.jobKey(schedule.getName(), schedule.getCronGroup());
+    }
+
+    private static void assertJob(JobDetail jobDetail, SchedulerConfiguration.Schedule schedule) {
+        assertThat(jobDetail.getDescription()).isEqualTo(schedule.getDescription());
+        assertThat(jobDetail.getJobClass()).isEqualTo(schedule.getJobClass());
+        assertThat(jobDetail.getKey()).isEqualTo(JobKey.jobKey(schedule.getName(), schedule.getCronGroup()));
+    }
+
+    private static void assertTrigger(Trigger trigger, SchedulerConfiguration.Schedule schedule) {
+        assertThat(trigger).isEqualTo(newTrigger()
+            .withIdentity(schedule.getName(), schedule.getCronGroup())
+            .withDescription(schedule.getDescription())
+            .withSchedule(cronSchedule(schedule.getCron()))
+            .build());
+    }
+
+    private interface JobA extends Job {
```

Comment:
wonted to prove that tested object uses what is in configuration. If every schedule have Job.class then test does not prove that it uses class from config (it can be e.g hardcoded in a code to always use same class). But there is not big value in this, i can remove

Reply:
If there is reason for it being there then I think it is fine, just wasn't sure of it initially, maybe add a comment explaining